BugOsPos.Application/Locations/GetLocationByIdQuery.cs

```diff
@@ -0,0 +1,40 @@
+﻿using BugOsPos.Application.Common.Interfaces.Persistence;
+using BugOsPos.Domain.LocationAggregate;
+using BugOsPos.Domain.LocationAggregate.ValueObjects;
+using ErrorOr;
+using FluentValidation;
+using MediatR;
+using BugOsPos.Domain.Common.ErrorsCollection;
+
+namespace BugOsPos.Application.Locations;
+
+public sealed record GetLocationByIdQuery(int Id) : IRequest<ErrorOr<GetLocationByIdResult>>;
+
+public sealed record GetLocationByIdResult(Location Location);
+
+public sealed class GetLocationByIdValidator : AbstractValidator<GetLocationByIdQuery>
+{
+    public GetLocationByIdValidator()
+    {
+        RuleFor(x => x.Id).NotEmpty();
+    }
+}
+
+public sealed class GetLocationByIdQueryHandler : IRequestHandler<GetLocationByIdQuery, ErrorOr<GetLocationByIdResult>>
+{
+    private readonly ILocationRepository _locationRepository;
+
+    public GetLocationByIdQueryHandler(ILocationRepository locationRepository)
+    {
+        _locationRepository = locationRepository;
+    }
+
+    public async Task<ErrorOr<GetLocationByIdResult>> Handle(GetLocationByIdQuery request, CancellationToken cancellationToken)
+    {
+        var customer = await _locationRepository.GetLocationById(LocationId.New(request.Id));
+        if (customer is null)
+            return Errors.Location.NotFound;
+
```

Comment:
Ideally, I guess, we would make put into request location id and franchise id.

If we get a valid location, we check if it's the same franchise. I feel I forgot to include field GroupId in Domain/LocationAggregate/Location. If we have that groupId, we could call `var group = _groupRepository.GetById(Location.GroupId)`. Then check if `group.FranchiseId == request.FranchiseId`. Not sure if that's how it should be done. Decide on your own is it better to leave current state or not.